### PR TITLE
NEW:85928 add reminder emails

### PR DESCRIPTION
**Pull request by jessicamoorthy**
- setup reminder emails section in site config
- add job that checks if approvals are pending every user defined number of days and resends approval emails

### CODE REVIEW

File: src/Extensions/SDLTSiteConfigExtension.php [hunk 1]
```diff
@@ -48,6 +49,7 @@ class SDLTSiteConfigExtension extends DataExtension implements ScaffoldingProvid
         'DataExportEmailSubject' => 'Text',
         'DataExportEmailBody' => 'HTMLText',
         'EmailSignature' => 'HTMLText',
+        'NumberOfDays' => 'Int',
```
Comment (roopamjain01) [inline src/Extensions/SDLTSiteConfigExtension.php:52]: can we please use more descriptive name of the field, right now it is generic?

Suggestion:- NumberODaysForApprovalReminderEmail

Comment (jessicamoorthy) [reply src/Extensions/SDLTSiteConfigExtension.php]: I agree, field name has been updated.

File: src/Job/CheckAwaitingApprovalsJob.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        $updatedQuestionnaireSubmission = QuestionnaireSubmission::get_by_id($this->questionnaireSubmission->ID);
+
+        if ($updatedQuestionnaireSubmission->QuestionnaireStatus == 'waiting_for_approval' )
+        {
+            $queuedJobService = QueuedJobService::create();
+
+            if($updatedQuestionnaireSubmission->isCisoApprovalPending()){
+                $queuedJobService->queueJob(
+                    new SendApprovalLinkEmailJob($this->questionnaireSubmission, $this->members),
```
Comment (roopamjain01) [inline src/Job/CheckAwaitingApprovalsJob.php:79]: Why are you createing new job here?
We should send email to ciso amd ba directly from here.

File: src/Job/CheckAwaitingApprovalsJob.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    public function process()
+    {
+        $siteConfig = SiteConfig::current_site_config();
+        if($siteConfig){
+           $numberOfDays = $siteConfig->NumberOfDaysForApprovalReminderEmail;
+        }
+        $updatedQuestionnaireSubmission = QuestionnaireSubmission::get_by_id($this->questionnaireSubmission->ID);
+
+        if ($updatedQuestionnaireSubmission->QuestionnaireStatus == 'waiting_for_approval' )
```
Comment (roopamjain01) [inline src/Job/CheckAwaitingApprovalsJob.php:74]: we need to send the email to both (ciso and bo), there is case when status is approved but ciso status will be pending.

File: src/Model/QuestionnaireSubmission.php [hunk 4]
```diff
@@ -1681,6 +1683,34 @@ public function resolve($object, array $args, $context, ResolveInfo $info)
...
+     *
+     * @return void
+     */
+    public function sendReminderEmails()
+    {
+        if ($this->isCisoApprovalPending() || $this->isBOApprovalPending()){
+
+            $cisoMembers = $this
+            ->getApprovalMembersListByGroup(GroupExtension::ciso_group());
+
```
Comment (roopamjain01) [inline src/Model/QuestionnaireSubmission.php:1698]: As I mentioned we can optimize this function, we can pass cisoMembers list in this function when you call this function.

File: src/Model/QuestionnaireSubmission.php [hunk 5]
```diff
@@ -2086,6 +2116,8 @@ public function updateQuestionnaireOnApproveAndDenyByGroup($status, $skipBoAndCi
                             date('Y-m-d H:i:s', time() + 90)
                         );
                     }
+                    $this->sendReminderEmails();
+
```
Comment (roopamjain01) [inline src/Model/QuestionnaireSubmission.php:2120]: you can call this function inside if condition.

File: src/Model/QuestionnaireSubmission.php [hunk 6]
```diff
@@ -1681,6 +1683,34 @@ public function resolve($object, array $args, $context, ResolveInfo $info)
...
+            $cisoMembers = $this
+            ->getApprovalMembersListByGroup(GroupExtension::ciso_group());
+
+            $siteConfig = SiteConfig::current_site_config();
+            if($siteConfig) {
+                $numberOfDays = $siteConfig->NumberOfDaysForApprovalReminderEmail;
+            }
+
+            if (!empty($cisoMembers) || !empty($businessOwnerEmail)) {
```
Comment (roopamjain01) [inline src/Model/QuestionnaireSubmission.php:1704]: If you call this function inside if condition then we don't need to check this condition again here.

**Pull request closed without merging**